async/chan max size reached during diagnostics processing

**Describe the bug**
Formatting a project with 50k lines is working, but with a lot of Java error logs.
The `Exception in thread "async-dispatch-2" java.lang.AssertionError:` log described in output is repeated 24 times before the normal logs related to format diff.

**Expected behavior**
To validate and format code without Java exception logs.

<details>
 <summary><b>clojure-lsp output</b></summary>
<pre>
Picked up _JAVA_OPTIONS: -Djava.locale.providers='COMPAT,JRE,CLDR'

[  0%] clojure-lsp                 
[  5%] Finding kondo config        
[ 10%] Finding cache               
[ 15%] Discovering classpath       
[ 20%] Copying kondo configs       
[ 90%] Resolving config paths      
[ 95%] Analyzing project files     
[100%] Project analyzed            
Formatting namespaces...
Exception in thread "async-dispatch-2" java.lang.AssertionError: Assert failed: No more than 1024 pending puts are allowed on a single channel. Consider using a windowed buffer.
(< (.size puts) impl/MAX-QUEUE-SIZE)
	at clojure.core.async.impl.channels.ManyToManyChannel.put_BANG_(channels.clj:156)
	at clojure.core.async.impl.ioc_macros$put_BANG_.invokeStatic(ioc_macros.clj:997)
	at clojure.core.async.impl.ioc_macros$put_BANG_.invoke(ioc_macros.clj:996)
	at clojure_lsp.feature.diagnostics$async_publish_diagnostics_BANG_$fn__33585$state_machine__7355__auto____33590$fn__33592.invoke(diagnostics.clj:121)
	at clojure_lsp.feature.diagnostics$async_publish_diagnostics_BANG_$fn__33585$state_machine__7355__auto____33590.invoke(diagnostics.clj:120)
	at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:978)
	at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:977)
	at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:982)
	at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:980)
	at clojure_lsp.feature.diagnostics$async_publish_diagnostics_BANG_$fn__33585.invoke(diagnostics.clj:120)
	at [clojure.lang.AFn.run](http://clojure.lang.afn.run/)([AFn.java:22](http://afn.java:22/))
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker([ThreadPoolExecutor.java:1128](http://threadpoolexecutor.java:1128/))
	at java.base/java.util.concurrent.ThreadPoolExecutor$[Worker.run](http://worker.run/)([ThreadPoolExecutor.java:628](http://threadpoolexecutor.java:628/))
	at clojure.core.async.impl.concurrent$counted_thread_factory$reify__1130$fn__1131.invoke(concurrent.clj:29)
	at [clojure.lang.AFn.run](http://clojure.lang.afn.run/)([AFn.java:22](http://afn.java:22/))
	at java.base/java.lang.Thread.run([Thread.java:834](http://thread.java:834/))
</pre>
</details>

**User details (please complete the following information):**
 - OS: MacOS
 - Version:  `clojure-lsp 2022.04.18-00.59.32-hotfix2` `clj-kondo 2022.04.09-SNAPSHOT`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,9 @@
   - Fix find definition of macros required by :require-macros on cljs files. #980
   - Add support to completion keywords found on function definition `:keys` destructuring, improving completion on api calls.
   - Avoid duplicate parens when inserting snippets during completion. #982
+  
+- CLI/API
+  - Make all cli features open files faster, improving speed of all CLI/API features. Fixes #985
 
 ## 2022.05.03-12.35.40
 

--- a/lib/src/clojure_lsp/feature/file_management.clj
+++ b/lib/src/clojure_lsp/feature/file_management.clj
@@ -31,12 +31,13 @@
                                :new-text new-text}]}]]
         (shared/client-changes changes db)))))
 
+(defn load-document! [uri text db*]
+  (swap! db* #(assoc-in % [:documents uri] {:v 0 :text text :saved-on-disk false})))
+
 (defn did-open [uri text db* allow-create-ns]
   (when-let [kondo-result (lsp.kondo/run-kondo-on-text! text uri db*)]
-    (swap! db* (fn [state-db]
-                 (-> state-db
-                     (assoc-in [:documents uri] {:v 0 :text text :saved-on-disk false})
-                     (lsp.kondo/db-with-results kondo-result))))
+    (load-document! uri text db*)
+    (swap! db* #(lsp.kondo/db-with-results % kondo-result))
     (f.diagnostic/async-publish-diagnostics! uri @db*))
   (when allow-create-ns
     (when-let [create-ns-edits (create-ns-changes uri text @db*)]

--- a/lib/src/clojure_lsp/feature/format.clj
+++ b/lib/src/clojure_lsp/feature/format.clj
@@ -2,6 +2,7 @@
   (:require
    [cljfmt.core :as cljfmt]
    [cljfmt.main :as cljfmt.main]
+   [clojure-lsp.feature.file-management :as f.file-management]
    [clojure-lsp.parser :as parser]
    [clojure-lsp.refactor.edit :as edit]
    [clojure-lsp.settings :as settings]
@@ -39,9 +40,9 @@
 (def cljfmt-config
   (memoize/ttl resolve-cljfmt-config :ttl/threshold memoize-ttl-threshold-milis))
 
-(defn formatting [uri db]
-  (let [{:keys [text]} (get-in db [:documents uri])
-        cljfmt-settings (cljfmt-config db)
+(defn formatting [uri db*]
+  (let [text (f.file-management/force-get-document-text uri db*)
+        cljfmt-settings (cljfmt-config @db*)
         new-text (cljfmt/reformat-string text cljfmt-settings)]
     (if (= new-text text)
       []

--- a/lib/src/clojure_lsp/handlers.clj
+++ b/lib/src/clojure_lsp/handlers.clj
@@ -384,7 +384,7 @@
 (defn formatting [{:keys [textDocument]}]
   (shared/logging-task
     :formatting
-    (f.format/formatting textDocument @db/db*)))
+    (f.format/formatting textDocument db/db*)))
 
 (defn range-formatting [{:keys [textDocument range]}]
   (process-after-changes

--- a/lib/src/clojure_lsp/internal_api.clj
+++ b/lib/src/clojure_lsp/internal_api.clj
@@ -239,7 +239,7 @@
   (filter :uri ns+uris))
 
 (defn ^:private open-file! [{:keys [uri] :as ns+uri} components]
-  (handlers/did-open {:textDocument {:uri uri :text (slurp uri)}} components)
+  (f.file-management/load-document! uri (slurp uri) (:db* components))
   ns+uri)
 
 (defn ^:private find-new-uri-checking-rename

--- a/lib/test/clojure_lsp/features/format_test.clj
+++ b/lib/test/clojure_lsp/features/format_test.clj
@@ -14,18 +14,18 @@
   (testing "when custom config file doesn't exists"
     (with-redefs [shared/file-exists? (constantly false)]
       (is (= "(a)\n(b c d)"
-             (:new-text (first (f.format/formatting (h/file-uri "file:///a.clj") @db/db*)))))))
+             (:new-text (first (f.format/formatting (h/file-uri "file:///a.clj") db/db*)))))))
   (testing "when custom config file exists"
     (with-redefs [shared/file-exists? (constantly true)
                   slurp (constantly "{}")]
       (is (= "(a)\n(b c d)"
-             (:new-text (first (f.format/formatting (h/file-uri "file:///a.clj") @db/db*))))))))
+             (:new-text (first (f.format/formatting (h/file-uri "file:///a.clj") db/db*))))))))
 
 (deftest test-formatting-noop
   (swap! db/db* shared/deep-merge {:project-root-uri (h/file-uri "file:///project")})
   (h/load-code-and-locs "(a)\n(b c d)")
   (with-redefs [shared/file-exists? (constantly false)]
-    (let [r (f.format/formatting (h/file-uri "file:///a.clj") @db/db*)]
+    (let [r (f.format/formatting (h/file-uri "file:///a.clj") db/db*)]
       (is (empty? r))
       (is (vector? r)))))
 

